### DISCUSSION TITLE
Terminate stream prior to end of the GCODE

### DISCUSSION THREAD

**Discussion by vaderag** [FormerLurker/Octolapse#701] [Q&A] [answered]
I'm using the Tuya plugin which which turns off my printer on the last line of my end code (M81 ipaddress), but it seems that because it's the last line of the gcode Octolapse think that the print failed (as it didn't get to the 'end' of the gcode presumably)

Is there any way (without going fully manual gcode) to tell octolapse to finish early (in my end code perhaps) so that the last line doesnt turn off the printer and cause octolapse to think that the print failed?

Thanks

**Comment by FormerLurker** [comment, marked as answer]
YES!  you can use the ```@OCTOLAPSE STOP-SNAPSHOTS``` and  ```@OCTOLAPSE START-SNAPSHOTS``` command.  Put them anywhere in your start/end/layer change gcode where you want octolapse to stop or start detecting snapshot positions.  This is also useful in the start/end gcode to prevent Octolapse from taking snapshots while priming, or while performing end of print moves.
